src/v/cluster/tm_stm.h

```diff
@@ -66,6 +61,42 @@ class tm_stm final : public persisted_stm {
         timeout
     };
 
+    struct tm_tx_hash_ranges {
+        using hash_ranges_type = std::vector<std::pair<uint32_t, uint32_t>>;
+        static constexpr uint8_t version = 0;
+
+        hash_ranges_type hash_ranges{};
```

Comment:
we need not only the intervals but the dots (individual tx ids) too

Reply:
done